ArduPlane/GCS_Mavlink.cpp

```diff
@@ -794,13 +794,169 @@ MAV_RESULT GCS_MAVLINK_Plane::handle_command_int_do_reposition(const mavlink_com
     return MAV_RESULT_FAILED;
 }
 
+// these are GUIDED mode commands that are RATE or slew enabled, so you can have more powerful control than default controls.
+MAV_RESULT GCS_MAVLINK_Plane::handle_command_int_guided_slew_commands(const mavlink_command_int_t &packet)
+{
+  switch(packet.command) {
+    
+#if OFFBOARD_GUIDED == ENABLED
+    case MAV_CMD_GUIDED_CHANGE_SPEED: {
+        // command is only valid in guided mode
+        if (plane.control_mode != &plane.mode_guided) {
+            return MAV_RESULT_TEMPORARILY_REJECTED;
+        }
+
+         // only airspeed commands are supported right now...
+        if (int(packet.param1) != SPEED_TYPE_AIRSPEED) {  // since SPEED_TYPE is int in range 0-1 and packet.param1 is a *float* this works.
+            return MAV_RESULT_UNSUPPORTED;
+        }
+
+         // reject airspeeds that are outside of the tuning envelope
+        if (packet.param2 > plane.aparm.airspeed_max || packet.param2 < plane.aparm.airspeed_min) {
+            return MAV_RESULT_UNSUPPORTED;
+        }
+
+         // reject duplicate airspeeds
+        float new_target_airspeed_cm = packet.param2 * 100;
+//        if (new_target_airspeed_cm == plane.guided_state.target_airspeed_cm) {
+        if ( fabsf(new_target_airspeed_cm - plane.guided_state.target_airspeed_cm)<0.001f) { // compare if two floats are 'the same'
```

Comment:
we use is_equal() in ArduPilot, unless there is some reason for this to be a special threshold. Also, please remove the commented code above

Reply:
will do.